Library/Homebrew/cmd/update.sh

```diff
@@ -384,7 +384,8 @@ EOS
   fi
 
   if ! git --version &>/dev/null ||
-       [[ -n "$HOMEBREW_SYSTEM_GIT_TOO_OLD" &&
+     [[ ( -n "$HOMEBREW_SYSTEM_GIT_TOO_OLD" ||
```

Comment:
Given you always set force brewed git above when git is too old you can just make this check for forced brewed git

Reply:
Oh, above = brew.sh? We can count on that always running first?
I suppose that's what the CURL options are doing, so I'll match that.